### PR TITLE
7_5_X_AlCaPhiSym_ringCut

### PR DESCRIPTION
**Pull request by bmarzocc**
Possibility to put different ring dependent cut on ADC

### CODE REVIEW

File: HLTrigger/special/src/HLTEcalPhiSymFilter.cc [hunk 1]
```diff
@@ -43,11 +49,19 @@ HLTEcalPhiSymFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptio
...
+                                                   6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,
+                                                   6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.,6.});
+  desc.add<std::vector<double> >("ampCut_endcapP",{9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,
+                                                   9.,9.,9.,9.,9.});
+  desc.add<std::vector<double> >("ampCut_endcapM",{9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,9.,
+                                                   9.,9.,9.,9.,9.});
   desc.add<std::string>("phiSymBarrelDigiCollection","phiSymEcalDigisEB");
   desc.add<std::string>("phiSymEndcapDigiCollection","phiSymEcalDigisEE");
-  descriptions.add("alCaPhiSymStream",desc);
+  descriptions.add("AlCaPhiSymStream",desc);
```
Comment (fwyzard) [inline HLTrigger/special/src/HLTEcalPhiSymFilter.cc]: the default name should be `hltEcalPhiSymFilter`

Comment (bmarzocc) [inline HLTrigger/special/src/HLTEcalPhiSymFilter.cc]: It has always been that name, I've only changed the capital letter. If needed I can change it to be hltEcalPhiSymFilter.

Comment (fwyzard) [inline HLTrigger/special/src/HLTEcalPhiSymFilter.cc]: Please do.

Comment (bmarzocc) [inline HLTrigger/special/src/HLTEcalPhiSymFilter.cc]: Done

**Comment by Martin-Grunewald** [comment]
Make sure fillDescriptions sets the new parametrs so as to reproduce the old behaviour
as used so far in the actual HLT menu instances.

**Comment by fwyzard** [comment]
So, the defaults in fillDescriptions should be {8., 8., 8, ... } and {12.,
12., 12., ...} , to match the old defaults.

.A

On 25 May 2015 at 12:28, Martin Grunewald <notifications@github.com> wrote:

> Make sure fillDescriptions sets the new parametrs so as to reproduce the
> old behaviour
> as used so far in the actual HLT menu instances.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9232#issuecomment-105200697>.
>



-- 
Strategy is a system of expedients.

Field Marshall Helmuth Carl Bernard Graf von Moltke

**Comment by Martin-Grunewald** [comment]
For the 75X PR, also modify customizeHLTforCMSSW.py in HLTrigger/Configuration/python as follows:
add
```
# Possibility to put different ring dependent cut on ADC (PR #9232)                                                              
def customiseFor9232(process):
    if hasattr(process,'hltEcalPhiSymFilter'):
        if hasattr(process.hltEcalPhiSymFilter,'ampCut_barrel'):
            delattr(process.hltEcalPhiSymFilter,'ampCut_barrel')
        if hasattr(process.hltEcalPhiSymFilter,'ampCut_endcap'):
            delattr(process.hltEcalPhiSymFilter,'ampCut_endcap')
    return process
```
and call it:
```

# CMSSW version specific customizations                                                                                          
def customiseHLTforCMSSW(process,menuType="GRun",fastSim=False):
    import os
    cmsswVersion = os.environ['CMSSW_VERSION']

    if cmsswVersion >= "CMSSW_7_5":
        process = customiseFor9232(process)
        process = customiseFor8679(process)
        process = customiseFor8356(process)
        process = customiseFor7966(process)
        process = customiseFor7794(process)

    return process
```

**Comment by Martin-Grunewald** [comment]
The reason is that this change will first enter 75X (ie, the 75X PR)
and then eventually be backported to 74X, and our menus for both
74X and 75X are 74X-based!

**Comment by bmarzocc** [comment]
Everything modified as requested. Any news?

**Comment by Martin-Grunewald** [comment]
please test

**Comment by bmarzocc** [comment]
I have compilation trouble of the plain CMSSW_7_5_X_2015-05-27-2300/HLTrigger/special without this PR-merging.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**